### PR TITLE
fix(persistence): fix bugs when running w/previousRedis

### PR DESCRIPTION
**Pull request by asher**
- fix bugs that could break pipelines running via previousRedis during a migration
- add related test coverage, fix test issues if running with compile time @Nonnull checks (intellij)

### CODE REVIEW

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy [hunk 1]
```diff
@@ -94,10 +95,12 @@ abstract class ExecutionRepositoryTck<T extends ExecutionRepository> extends Spe
     def runningExecution = orchestration {
       status = RUNNING
       buildTime = 0
+      trigger = new ManualTrigger(null, "fnord", [:], null, null)
```
Comment (robfletcher) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy]: Hail Eris!

File: orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy [hunk 2]
```diff
@@ -566,16 +574,103 @@ class JedisExecutionRepositorySpec extends ExecutionRepositoryTck<JedisExecution
     100   || 4
   }
 
+  def "can store an orchestration already in previousRedis back to previousRedis"() {
```
Comment (robfletcher) [inline orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy]: Is that actually what we want to do? Don't we want to migrate it to the new Redis?

Comment (asher) [reply orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy]: Migration is still the work of `MultiRedisOrchestrationMigrationNotificationAgent` and `MultiRedisPipelineMigrationNotificationAgent`, which only operate over completed executions. We'll want to rewrite the migration process once `DynomiteExecutionRepository` is merged in with schema changes and I'd like to tackle migrating running executions then too.

Comment (robfletcher) [reply orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy]: Ok, cool

**Pull request merged**